### PR TITLE
Correct most of Rubocop offenses

### PR DESCRIPTION
**Pull request by syndbg**
Corrected most of Rubocop offenses automagically with `rubocop -a` and by hand.

There are still some line lengths offenses in comments and long string lines that I don't mark as needed to fix.

And there's the `app/workers/homebrew_formula_import_worker.rb` that offends ABC and MethodLength cops. Still I would avoid changing it.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -29,23 +31,23 @@ end
...
   gem 'better_errors'
-  gem 'binding_of_caller', :platforms=>[:mri_19, :mri_20, :rbx]
-  gem 'hub', :require => nil
+  gem 'binding_of_caller', platforms: [:mri_19, :mri_20, :rbx]
+  gem 'hub', require: nil
   gem 'rails_layout'
   gem 'yard'
   gem 'rails_best_practices'
-  gem 'rubocop'
+  gem 'rubocop', require: false
```
Comment (syndbg) [inline Gemfile]: https://github.com/bbatsov/rubocop#installation

File: app/models/homebrew/tools.rb [hunk 2]
```diff
@@ -58,7 +58,7 @@ def self.get_conflicts(formula)
 
       conflicts.gsub!(/('|")/, '')
 
-      conflicts, _ = extract_conflict_reason_if_possible!(conflicts)
+      conflicts, = extract_conflict_reason_if_possible!(conflicts)
```
Comment (syndbg) [inline app/models/homebrew/tools.rb]: this is actually valid syntax

File: app/models/software_description_fetchers/strategies/default.rb [hunk 3]
```diff
@@ -29,10 +30,10 @@ def clean_text(text)
       end
 
       def regex
-        /(^.*(?:#{@software_name}|#{@software_filename})
+        %r{(^.*(?:#{@software_name}|#{@software_filename})
           (?:\)|\s\u2122|\s[\d\.]+|\scodec)?\s(?:is\s
           (?:an?|the)|(?:project\s)?provides)[\s\w\'\(\)\,\-\+\/\.\:]+\.(?:\s|$)
-        )/ix
+        )}ix
```
Comment (syndbg) [inline app/models/software_description_fetchers/strategies/default.rb]: Could you carefully review this? I did not find a failing test, but I'm not 100% either that `ix` is respected

**Comment by zedtux** [comment]
Thank you @syndbg for this too :)

**Pull request merged**